pkg/thing/interactors/register_thing.go

```diff
@@ -87,21 +83,21 @@ func (i *ThingInteractor) verifyThingID(id string) error {
 	return nil
 }
 
-func (i *ThingInteractor) reply(id, token string, err error) error {
-	var errStr *string
-
-	if err != nil {
-		errStr = new(string)
-		*errStr = err.Error()
-	} else {
-		errStr = nil
+func (i *ThingInteractor) sendResponse(id, token string, err error) error {
+	sendErr := i.clientPublisher.SendRegisteredDevice(id, token, getErrMessagePtr(err))
+	if sendErr != nil {
+		if err != nil {
+			return fmt.Errorf("error sending response to client: %v: %w", sendErr, err)
+		}
+		return fmt.Errorf("error sending response to client: %v", sendErr)
 	}
+	return err
+}
 
-	errPublish := i.clientPublisher.SendRegisteredDevice(id, token, errStr)
-	if errPublish != nil {
-		i.logger.Error(errPublish)
-		return errPublish
+func getErrMessagePtr(err error) *string {
+	if err != nil {
+		msg := err.Error()
+		return &msg
 	}
-
 	return nil
 }
```

Comment:
I was actually wondering if this is the most idiomatic way to get values from repetitive clauses and apparently it is. Just leaving this as a +1 here :smile: 
> golang: https://golang.org/doc/faq#Does_Go_have_a_ternary_form
> stackoverflow: https://stackoverflow.com/questions/19979178/what-is-the-idiomatic-go-equivalent-of-cs-ternary-operator

Reply:
In terms of clean code, the better approach is to break that function. It was so hard to understand and it did more than one thing (despite being simple or not). At least for me, with `getErrMessagePtr()` the function calling it became clearer.